13369 Added pco2a_a_sample parser,drivers,test files to support new pco2a instrument

tlfish3638 opened the pull request.

danmergens commented on RELEASE_NOTES.md:
```diff
@@ -1,3 +1,7 @@
+# Version 0.7.2
```
interface change - use 0.8.0

tlfish3638 replied:
Done

danmergens commented on mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_recovered_driver.py:
```diff
@@ -0,0 +1,20 @@
+from mi.core.versioning import version
+
+
+@version("15.6.2")
```
If this is a new version, it should be 0.1.0 or something like that. Looks like you copied this from another driver.

tlfish3638 replied:
I dropped the recovered,telemetered drivers, so moved the parse function into the main driver and modified the version to 0.1.0 there.

danmergens commented on mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_recovered_driver.py:
```diff
@@ -0,0 +1,20 @@
+"""
+@package mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_recovered_driver
+@file mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_recovered_driver.py
+@author Sung Ahn
```
are you Sung?

tlfish3638 replied:
I am not

danmergens commented on mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_telemetered_driver.py:
```diff
@@ -0,0 +1,20 @@
+"""
+@package mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_telemetered_driver
+@file mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_telemetered_driver.py
+@author Sung Ahn
```
take credit

tlfish3638 replied:
OK!

danmergens commented on mi/dataset/parser/pco2a_a_sample.py:
```diff
@@ -0,0 +1,268 @@
+class DataParticleType(BaseEnum):
+    PCO2A_INSTRUMENT_AIR_PARTICLE = 'pco2a_a_dcl_instrument_air'
+    PCO2A_INSTRUMENT_WATER_PARTICLE = 'pco2a_a_dcl_instrument_water'
+#   PCO2A_INSTRUMENT_AIR_RECOVERED_PARTICLE = 'pco2a_a_dcl_instrument_air_recovered'
```
ok to remove - maybe put in a comment that the recovered streams will use the same stream name if you want.

tlfish3638 replied:
Done

danmergens commented on mi/dataset/parser/pco2a_a_sample.py:
```diff
@@ -0,0 +1,268 @@
+        self.contents[DataParticleKey.PREFERRED_TIMESTAMP] = DataParticleKey.PORT_TIMESTAMP
+
+
+class Pco2aADclTelemeteredInstrumentDataParticleAir(Pco2aADclInstrumentDataParticleAir):
```
I think you only need `Pco2aADclInstrumentDataParticleAir` and then point the references of both the telemetered and recovered driver to the one class.

tlfish3638 replied:
Done

tlfish3638 commented:
I cleaned things up a fair amount to eradicate the recovered,telemetered drivers and classes, removed unnecessary commented lines, etc. I retested the unit test file and re-ingested the data successfully. Give it another review. Thanks.

danmergens commented on mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_driver.py:
```diff
@@ -0,0 +1,49 @@
+"""
+
+
+from mi.core.log import get_logger
```
`get_logger` is outdated. Instead of:
```python
from mi.core.log import get_logger
log = get_logger()
```
use:
```python
from mi.core.log import log
```

tlfish3638 replied:
Done

danmergens commented on mi/dataset/driver/pco2a_a/sample/pco2a_a_sample_driver.py:
```diff
@@ -0,0 +1,49 @@
+    Pco2aADclInstrumentDataParticleAir, \
+    Pco2aADclInstrumentDataParticleWater
+
+log = get_logger()
```
see comment above

tlfish3638 replied:
Done

danmergens commented on mi/dataset/parser/pco2a_a_sample.py:
```diff
@@ -0,0 +1,232 @@
+
+import re
+
+from mi.core.log import get_logger
```
use `import log` instead

tlfish3638 replied:
Done

danmergens commented on mi/dataset/parser/pco2a_a_sample.py:
```diff
@@ -0,0 +1,232 @@
+SENSOR_DATA_PATTERN_WATER += SENSOR_DATA_PATTERN
+SENSOR_DATA_MATCHER_WATER = re.compile(SENSOR_DATA_PATTERN_WATER)
+
+print(SENSOR_DATA_PATTERN_WATER)
```
debug - please remove

tlfish3638 replied:
Done

danmergens commented on mi/dataset/parser/pco2a_a_sample.py:
```diff
@@ -0,0 +1,232 @@
+
+print(SENSOR_DATA_PATTERN_WATER)
+
+# Manual test is below
```
move this into the unit test class

tlfish3638 replied:
Done

sgfoote commented on mi/dataset/parser/test/test_pco2a_a_sample.py:
```diff
@@ -0,0 +1,256 @@
+     MODULE_NAME, PARTICLE_CLASSES
+#    MODULE_NAME, RECOVERED_PARTICLE_CLASSES, TELEMETERED_PARTICLE_CLASSES
+
+log = get_logger()
```
Use of deprecated get_logger.

tlfish3638 replied:
Done
